include/boost/circular_buffer/base.hpp

```diff
@@ -1460,7 +1460,8 @@ private empty_value<Alloc>
         if (full()) {
             if (empty())
                 return;
-            replace(m_last, value_type(::boost::forward<Args>(args)...));
```

Comment:
this line is wrong. This is your bug!

This is where the temporary object gets created.

What do you think value_type(...) is doing?

It creates a temporary object and call the replace() method that is using assignment!